app/Models/Transceiver.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use LibreNMS\Interfaces\Models\Keyable;
+
+class Transceiver extends PortRelatedModel implements Keyable
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'device_id',
+        'port_id',
+        'index', // sfp index to identify this sfp uniquely by snmp data
+        'type', // module type sfp, xfp, sfp+, qsfp, sfp28, etc or more detailed info like 10G_BASE_SR_SFP
+        'vendor', // vendor name
+        'oui', // vendor oui
+        'model', // model number or name
+        'revision', // hardware revision
+        'serial', // serial number
+        'date', // date of manufacture
+        'ddm', // if the module supports DDM or DOM
+        'encoding', // data encoding method
+        'cable', // SM, MM, Copper, etc
+        'distance', // Max distance or measured distance
+        'wavelength', // Middle Wavelength in nanometers
+        'connector', // LC, SM, RJ45, etc physical connector type, See Ocnos for some normalized connectors
```

Comment:
Wondering if we should enforce the values, there aren't that many connectors. Makes sorting etc nicer.

Reply:
I counted about 30 connector types :D

Reply:
https://en.wikipedia.org/wiki/Optical_fiber_connector

Reply:
We can probably make them consistent by convention instead of enforcing.

Reply:
Would suggest to stick to the lists/naming conventions used in the [SFF-8024 specification](https://members.snia.org/document/dl/26423):
![image](https://github.com/librenms/librenms/assets/1706402/e912df6d-8c8f-408c-a058-5f10f80d4463)
And odd version returned by switch vendors should be replaced with the standard naming but I wouldn't enforce specific values to allow easy support of new connectors.